test/scripts/console/generate.js

```diff
@@ -168,4 +168,67 @@ describe('generate', () => {
     result[0].should.eql('bb');
     result[1].should.eql('cc');
   }));
+
+  it('proceeds after error when bail option is not set', () => {
+    hexo.extend.renderer.register('err', 'html', () => Promise.reject(new Error('Testing unhandled exception')));
```

Comment:
Will the error populate in this situation?
```
hexo.extend.renderer.register('syncError', 'html', () => { 
throw new Error('Testing unhandled exception')
});
```

Reply:
Specs are green when I replace rejected promise with `throw new Error`
```
hexo.extend.renderer.register('err', 'html', () => {
      throw new Error('Testing unhandled exception')
    });
```

I'm not sure what did you mean by `syncError`, with the following code
```
hexo.extend.renderer.register('syncError'
```
that callback is never called.